src/rust/ide/ast/impl/src/opr.rs

```diff
@@ -219,30 +276,117 @@ impl Chain {
     /// If this is infix, it flattens whole chain and returns result.
     /// Otherwise, returns None.
     pub fn try_new(ast:&Ast) -> Option<Chain> {
-        GeneralizedInfix::try_new_root(&ast).map(|infix| infix.flatten())
+        GeneralizedInfix::try_new(&ast).map(|infix| infix.flatten())
     }
 
     /// Flattens infix chain if this is infix application of given operator.
     pub fn try_new_of(ast:&Ast, operator:&str) -> Option<Chain> {
-        let infix = GeneralizedInfix::try_new_root(&ast)?;
+        let infix = GeneralizedInfix::try_new(&ast)?;
         (infix.name() == operator).as_some_from(|| infix.flatten())
     }
 
     /// Iterates over &Located<Ast>, beginning with target (this argument) and then subsequent
     /// arguments.
-    pub fn enumerate_operands<'a>(&'a self) -> impl Iterator<Item=&'a Shifted<Located<Ast>>> + 'a {
-        let this = std::iter::once(&self.target);
-        let args = self.args.iter().map(|elem| &elem.operand);
-        this.chain(args).flatten()
+    pub fn enumerate_operands<'a>(&'a self) -> impl Iterator<Item=Located<&'a Shifted<Ast>>> + 'a {
+        let this_crumbs = self.args.iter().rev().map(ChainElement::crumb_to_previous).collect_vec();
+        let this        = self.target.as_ref().map(|opr| Located::new(this_crumbs,opr));
+        let args        = self.args.iter().enumerate().map(move |(i,elem)| elem.operand.as_ref().map(|opr| {
+            let to_infix   = self.args.iter().skip(i+1).rev().map(ChainElement::crumb_to_previous);
```

Comment:
could you make `i+1` as variable and describe what it means here please? Im trying to follow this chain of things but its not easy